### ISSUE TITLE
Clarify Semantics for Statistics

### ISSUE THREAD

**Issue by yannis666** [jsr107/jsr107spec#50]
A while ago Statistics were introduced to the spec. We are in the process of tightening up the specification, and want to make sure that the exact meaning of the statistics are defined. Somewhat similarly to the decisions to be made for read through there can be different interpretations of when counters should be incremented.

I have added a [table](https://docs.google.com/document/d/1YZ-lrH6nW871Vd9Z34Og_EqbX_kxxJi55UrSn4yL2Ak/edit?hl=en_US&pli=1#heading=h.pr5diw9wshmt) to the spec to help drive the discussion. The table shows the behavior as exists in the RI currently and potential changes.

Examples of possible changes are 
- if `containsKey` modifies counts (It does in Coherence but not in the RI)
- if `getAndPut` increments the get count (does not in RI)

Basically there are a number of API methods that implicitly do a "get like" operation.
